vast/CMakeLists.txt

```diff
@@ -18,19 +14,26 @@ install(
 
 macro (create_tenzir_symlink name)
   add_custom_command(
-    OUTPUT "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}/${name}"
+    TARGET tenzir POST_BUILD
     COMMAND ${CMAKE_COMMAND} -E create_symlink "./tenzir"
-            "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}/${name}"
-    DEPENDS tenzir
-    COMMENT "Creating ${name} symlink")
-  add_custom_target(
-    "${name}" ALL DEPENDS "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}/${name}")
-  install(
-    CODE "
-    execute_process(COMMAND
-      \"${CMAKE_COMMAND}\" -E create_symlink \"./tenzir\"
-      \"\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}/${name}\")"
-    COMPONENT Runtime)
+            "$<TARGET_FILE_DIR:tenzir>/${name}"
+    COMMENT "Creating ${name} symlink"
+    VERBATIM)
+  if (VAST_ENABLE_RELOCATABLE_INSTALLATIONS)
+    install(
+      CODE "
+      execute_process(COMMAND
+        \"${CMAKE_COMMAND}\" -E create_symlink \"./tenzir\"
+        \"\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}/${name}\")"
+      COMPONENT Runtime)
+  else ()
+    install(
+      CODE "
+      execute_process(COMMAND
+        \"${CMAKE_COMMAND}\" -E create_symlink \"./tenzir\"
+        \"\$ENV{DESTDIR}${CMAKE_INSTALL_FULL_BINDIR}/${name}\")"
+      COMPONENT Runtime)
```

Comment:
This should not be needed. This code should make sure that we can never run into this situation: https://github.com/tenzir/tenzir/blob/58b2be734e0b1f348afcc987bfefb643d05f6761/CMakeLists.txt#L176-L191

Reply:
Should likely be `COMPONENT Runtime` though so this conflict would always be detected.

Reply:
Yes it is needed. We correctly only normalize install dirs in relocatable mode, so
`{CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}` contains the prefix twice, making it fail.

Reply:
I see. But that's still problematic, as it would similarly affect all other places in which we use `install(CODE …)`.

Do we have a good reason to support non-relocatable builds in the first place?

Reply:
We use it in non-static Nix builds, which this PR is intended to fix.

Reply:
If they could also be made relocatable builds then we could drop the CMake option, and would not need to have this complexity for all custom install scripts. Wouldn't that be a lot easier to maintain than fixing the issue?

Reply:
> Wouldn't that be a lot easier to maintain than fixing the issue?

The would need to investigated, and I won't spend the time on that now.
The provided patch works.

Reply:
I don't want to spend too much time on this, on that front I agree. But:
- We do not test this build configuration in CI
- It is a very confusing piece of code to understand

So the very least this PR should get is a comment that explains _why_ this logic exists, and also _why_ the install-time vs. build-time prefix mismatch detection logic does not suffice for this.

Reply:
That detection logic is not relevant for this. I added the requested comment.